### PR TITLE
[ci][release-infra] use python docker image for running release tests

### PR DESCRIPTION
**Pull request by can-anyscale**
Currently we are using a ray docker image to trigger release tests. This PR replaces that with just the python docker image. That is much cheaper and keep things simpler.

Also propagate the environment to the test job, which is configurable by a lot of different environment variable.

Test strategies:
- Release tests

### CODE REVIEW

File: release/ray_release/buildkite/step.py [hunk 1]
```diff
@@ -94,15 +85,8 @@ def get_step(
...
-
-    if "python" in test:
-        python_version = parse_python_version(test["python"])
-    else:
-        python_version = DEFAULT_PYTHON_VERSION
-
-    step["plugins"][0][DOCKER_PLUGIN_KEY][
-        "image"
-    ] = f"rayproject/ray:nightly-py{python_version_str(python_version)}"
+    step["plugins"][0][DOCKER_PLUGIN_KEY]["image"] = "python:3.8"
```
Comment (aslonnie) [inline release/ray_release/buildkite/step.py:88]: you can also use a digest (like `python@sha256:xxxxx`) to make it more deterministic.

**Review by aslonnie — APPROVED** [review]

**Review by krfricke — CHANGES_REQUESTED** [review]
Wait, this removes the full path for ray test repo/branch. This means you can't specify a different test repo + branch in the buildkite dialog anymore.

IMO the regular path in `build_pipeline.py` should remain. Happy to get rid of the re-running of the script and installing ray - we mostly needed that for ray client-related things.

**Comment by can-anyscale** [comment]
@krfricke got you, I'm trying to see if we really need this feature, what is the use case for providing a different test repo that is not rayproject/ray? Also the the branch, can someone create a buildkite build with that branch instead?

**Comment by krfricke** [comment]
> @krfricke got you, I'm trying to see if we really need this feature, what is the use case for providing a different test repo that is not rayproject/ray? Also the the branch, can someone create a buildkite build with that branch instead?

It was originally used to be able to manually kick off release tests on development branches that can live in forked repos. But the same code path is currently used to kick off release tests from PRs:

https://github.com/ray-project/ray/blob/master/release/ray_release/buildkite/settings.py#L154-L158

So without this path, we couldn't run changed or new release tests from PRs.

**Comment by can-anyscale** [comment]
Got you, is it ok to stop supporting the release test infra in a forked repo at this point then.

And for PR build, buildkite job should use the branch and commit from BUILDKITE_BRANCH and BUILDKITE_COMMIT and will work seamlessly right? Or did I miss something?

**Comment by krfricke** [comment]
> Got you, is it ok to stop supporting the release test infra in a forked repo at this point then.

That's fine. In that case we should remove [the dialog option](https://github.com/ray-project/buildkite-ci-stack/blob/master/ci-release-test-module/buildkite_steps.tpl.yaml#L38-L42) first.

> And for PR build, buildkite job should use the branch and commit from BUILDKITE_BRANCH and BUILDKITE_COMMIT and will work seamlessly right? Or did I miss something?

Ah I see. I think you're right. Maybe this was a remnant from more complicated times. Happy to go ahead then, ping me when the dialog option is removed!

**Review by krfricke — APPROVED** [review]

**Comment by krfricke** [comment]
Oh, and some tests are failing

File: release/ray_release/buildkite/step.py (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/buildkite/step.py:88]: I'll keep it as python:3.8 and trust their upgrades will not cause too much trouble for now ;)

**Comment by can-anyscale** [comment]
Fixed tests and remove the release test branch + repo dialog \o/

**Pull request merged**